### PR TITLE
Add Thanos service monitors

### PR DESCRIPTION
**Pull request by kakkoyun**
/cc @metalmatze @squat

### CODE REVIEW

**Comment by metalmatze** [comment]
Generally, lgtm. Let's wait until we merge this downstream. :blush:

File: environments/sre/servicemonitors/thanos-querier-serviceMonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+serviceMonitor:
```
Comment (metalmatze) [inline environments/sre/servicemonitors/thanos-querier-serviceMonitor.yaml]: We need to specifically get this object. So without that top level `serviceMonitor`.

File: environments/sre/main.jsonnet [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+              matchLabels: ['telemeter-stage'],
+            },
+          },
+        },
+      },
+    },
+  };
+
+{
+  'thanos-querier-serviceMonitor': sm.thanos.querier,
```
Comment (metalmatze) [inline environments/sre/main.jsonnet]:
```suggestion
  'thanos-querier-serviceMonitor': sm.thanos.querier.serviceMonitor,
```

**Comment by metalmatze** [comment]
Thanks for fixing the ServiceMonitors. Now you actually have a conflict in the `jsonnetfile.lock.json`. To resovled that, you can accept either of the hashes and then run `jb update` one more time. Then commit and push.

**Comment by kakkoyun** [comment]
@metalmatze It should be good now

File: environments/sre/main.jsonnet [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  {
+    thanos+:: {
+      querier+: {
+        serviceMonitor+: {
+          metadata: {
+            name: 'observatorium-thanos-querier',
+            labels: { prometheus: 'app-sre' },
+          },
+          spec+: {
+            selector+: {
```
Comment (squat) [inline environments/sre/main.jsonnet]: are we intentionally _extending_ the selector but _overriding_ the namespaceSelector?

Comment (kakkoyun) [reply environments/sre/main.jsonnet]: We do. Is this a mistake? I just wanted to add `namespaceSelector`

Comment (squat) [reply environments/sre/main.jsonnet]: no not necessarily a mistake, just wanted to know if it was intentional and why the reasoning is

**Pull request merged**